sympy/solvers/inequalities.py

```diff
@@ -432,10 +433,10 @@ def valid(x):
     for x in reals:
         end = x
 
-        if ((end is S.NegativeInfinity and expr.rel_op in ['>', '>=']) or
-           (end is S.Infinity and expr.rel_op in ['<', '<='])):
-            sol_sets.append(Interval(start, S.Infinity, True, True))
-            break
+        if end in [S.NegativeInfinity, S.Infinity]:
+            if valid(S(0)):
```

Comment:
The point of testing 0 is testing any arbitrary real (not extended real) to check if the inequality makes sense or not,
Since the issue was this:
`solve_univariate_inequality(x > -oo)` works but `solve_univariate_inequality(-oo < x)` doesn't. (Previous implementation only covers the first case)
When end is +/- oo , these equivalent situations arises:
1). `x > oo`
2). `x < oo`
3). `x > -oo`
4). `x < -oo`
Of which (1) & (4) doesn't makes sense, the test here is to return False for these 2 situations.
And other of them to return the `Interval(-oo, oo)` , irrespective of the way of writing the inequality.